lib/hooks/use-pinch.js

```diff
@@ -1,66 +1,50 @@
-import {
-	useEffect, useCallback, useReducer
-} from 'haunted';
-import {
-	pinchReducer, initState, PINCHING
-} from '../reducers/pinch';
-
-const getDistance = (ev1, ev2) => Math.hypot(ev2.pageX - ev1.pageX, ev2.pageY - ev1.pageY),
-
-	/**
-	 * Tracks the amount a user pinched the document.
-	 *
-	 * Tracking is started by calling the `start` function.
-	 *
-	 * @return {Array} [status: string, delta: number, start: function]
-	 */
-	usePinch = () => {
-		const [state, dispatch] = useReducer(pinchReducer, initState),
-
-			// TODO: handle variable number of touches a la use-touch-pan
-			start = useCallback(ev => {
-				if (ev.touches.length < 2) {
-					return;
-				}
-
-				dispatch({
-					type: 'start',
-					initialDistance: getDistance(...ev.touches)
-				});
-			}, [dispatch]),
-
-			pinch = useCallback(ev => {
-				dispatch({
-					type: 'pinch',
-					distance: getDistance(...ev.touches)
-				});
-			}, [dispatch]),
+import { useEffect } from 'haunted';
+import { pinchReducer } from '../reducers/pinch';
+
+const getDistance = (ev1, ev2) => Math.hypot(ev2.pageX - ev1.pageX, ev2.pageY - ev1.pageY);
+
+/**
+ * Tracks the amount a user pinched the document.
+ *
+ * Tracking is started by calling the `start` function.
+ *
+ * @return {Array} [isPinching: boolean, delta: number, start: function]
+ */
+export const usePinch = () => {
+	const pincher = pinchReducer(),
+
+		// TODO: handle variable number of touches a la use-touch-pan
+		start = ev => {
+			if (ev.touches.length < 2) {
+				return;
+			}
 
-			stop = useCallback(ev => {
-				if (ev.touches.length > 2) {
-					return;
-				}
+			pincher.start(getDistance(...ev.touches));
+		},
 
-				dispatch({ type: 'stop' });
-			}, [dispatch]);
+		pinch = ev => pincher.pinch(getDistance(...ev.touches)),
 
-		useEffect(() => {
-			if (state.status !== PINCHING) {
+		stop = ev => {
+			if (ev.touches.length > 2) {
 				return;
 			}
-			const opts = { passive: true };
 
-			document.addEventListener('touchmove', pinch, opts);
-			document.addEventListener('touchend', stop, opts);
-			return () => {
-				document.removeEventListener('touchmove', pinch, opts);
-				document.removeEventListener('touchend', stop, opts);
-			};
-		}, [state.status, stop, pinch]);
+			pincher.stop();
+		};
+
+	useEffect(() => {
+		if (!pincher.isPinching) {
+			return;
+		}
+		const opts = { passive: true };
 
-		return [state.status, state.delta, start];
-	};
+		document.addEventListener('touchmove', pinch, opts);
+		document.addEventListener('touchend', stop, opts);
+		return () => {
+			document.removeEventListener('touchmove', pinch, opts);
+			document.removeEventListener('touchend', stop, opts);
+		};
+	}, [pincher.isPinching, stop, pinch]);
```

Comment:
Because `stop` and `pinch` are new functions (you dropped useCallback) this effect runs always.